### PR TITLE
Code review latest changes

### PR DESCRIPTION
**Pull request by mcarans**
This "review" branch, which is one commit before "main", is purely for doing a code review of the changes already in the "main" branch. The PR should be closed once the code review is complete as there is nothing to merge into "main".

### CODE REVIEW

File: cods.py [hunk 1]
```diff
@@ -1,182 +1,134 @@
...
+            if not service_list:
+                self.errors.add(f"{iso}: could not find service list")
+                return resources
+
+            for service in service_list:
+                resource = dict()
+                if service["name"].split("/")[1][:3].upper() != iso:
+                    continue
+
+                resource["url"] = url + "/" + service["name"].split("/")[1] + "/" + service["type"]
```
Comment (mcarans) [inline cods.py]: I suggest making a variable for `service["name"].split("/")[1]` and using it in both places. Also I suggest `resource["url"] = f"{url}/MYVAR/{service['type']}"`

File: cods.py [hunk 2]
```diff
@@ -1,182 +1,134 @@
...
+                resource["name"] = f"{service['name']} ({service['type']})"
+                resource["format"] = "Geoservice"
+
+                try:
+                    resource_desc = self.downloader.download_json(resource["url"] + "?f=pjson")
```
Comment (mcarans) [inline cods.py]: Use the `f"..."` style

**Review by mcarans — COMMENTED** [review]
I think the easiest way to increase coverage would be to use the class hdx.utilities.retriever.Retrieve as described here: https://hdx-python-utilities.readthedocs.io/en/latest/#retrieving-files. You can also look at a scraper for usage of Retrieve for testing eg. https://github.com/OCHA-DAP/hdx-scraper-ifrc/blob/main/tests/test_ifrc.py#L50-L59

**Comment by b-j-mills** [comment]
Changed downloader to retriever, updated tests, and updated string formatting.
Coverage is now at 75% - thanks!

**Pull request closed without merging**